pkg/provider/aci.go

```diff
@@ -303,6 +305,20 @@ func (p *ACIProvider) CreatePod(ctx context.Context, pod *v1.Pod) error {
 		return err
 
 	}
+
+
+	// if no username credentials are provided use agentpool MI if for pulling images from ACR
+	if len(*creds) == 0 {
+		agentPoolKubeletIdentity, err := GetAgentPoolKubeletIdentity(ctx, p.resourceGroup, p.vnetSubscriptionID)
```

Comment:
Would you please have the `GetAgentPoolKubeletIdentity` function as a method instead?